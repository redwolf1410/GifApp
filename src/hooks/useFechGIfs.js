
import React ,{ useState,useEffect } from 'react'
import { getGifs } from "../helpers/getGifs";
export const useFechGifs = (category) => {
    const [state, setState] = useState({
        
        data: [],
        loading: true
    });
    useEffect(() => {

        getGifs(category).then(imgs => 
            setTimeout(()=>{
                setState({
                    data: imgs,
                    loading: false})
            },100)
            );

    }, [category  ])

    
    return state
}
