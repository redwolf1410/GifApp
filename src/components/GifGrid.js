import React,{useEffect,useState} from 'react'
import { GifGridItem } from './GifGridItem';
import { getGifs } from "../helpers/getGifs";
import { useFechGifs } from "../hooks/useFechGifs";
import PropTypes from "prop-types";
export const GifGrid = ({ category }) => {

    const {data:images, loading } = useFechGifs(category);

    return (
        <>
        <h3>{category}</h3>

        { loading ? 'Loading...' : 
         <div className='card-grid'>
            
            {
                images.map( img => (
                        <GifGridItem 
                        key={img.id}
                        {...img}/>     
                        )
                )
            }
        </div> }
        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}