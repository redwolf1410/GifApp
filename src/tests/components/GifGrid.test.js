import {GifGrid} from '../../components/GifGrid';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'
import React  from 'react';
import { useFechGifs } from '../../hooks/useFechGifs';
jest.mock('../../hooks/useFechGifs')

const value = 'One Punch'
describe('GifGrid tests', () => { 
    
    test('should be the same as the snapshot', () => { 

        useFechGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category = {value}/>)
        expect(wrapper).toMatchSnapshot();
        
     })
    test('should show items when images are loaded with useFetch', () => { 
        const gifs = [{
            id: 'ABC',
            url: 'http://localhost/tuputamaadre.jpg',
            title:'Cualquier cosa'
        }]
        useFechGifs.mockReturnValue({
            data: gifs,
            loading: true
        });
        const wrapper = shallow(<GifGrid category={value} />)
        expect(wrapper.find('p').exists()).toBe(false)        
     })

 })