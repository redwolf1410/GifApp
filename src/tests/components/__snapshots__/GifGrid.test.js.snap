// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid tests should be the same as the snapshot 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  Loading...
</Fragment>
`;
